### DISCUSSION TITLE
Please add support to composite key (and composite index)

### DISCUSSION THREAD

**Discussion by dbargaud** [directus/directus#22220] [Feature Requests]
### Summary

I know this is a long talk.
I've seen many discussions on this topic but composite key is mandatory in RDBMS system.

### Motivation

With a composite key we will be able to handle correctly M2M relationship (and uniqueness of the combination).

Right now it is really difficult to maintain some kind of datas (typically translations).
If you update an item with translations you must know each id for each language instead of simply knowing the language code.
This should always be a unique combination between the item and its translation).

Right know if you are not very careful it is easy to create incoherent datas by adding more than one time the same language.

It makes very difficult to maintain data coherence when you sync between two different systems (directus and other source).
You choose to use RDBMS instead of noSQL so, please, be compliant with the basics of RDBMS and make our life easier. :)

Directus is a great piece of software but not supporting composite primary key is a major flow.

### Detailed Design

don't know :)

### Requirements List

don't know :)

### Drawbacks

no breaking change 

### Alternatives

There is unfortunately no alternative.

### Adoption Strategy

There is no breaking change since this will affect only new collections / relationships.

**Comment by br41nslug** [comment]
Duplicate of https://github.com/directus/directus/discussions/12137
